src/app/server/BUILD.gn

```diff
@@ -36,14 +36,20 @@ static_library("server") {
     "CommissioningWindowManager.h",
     "DefaultAclStorage.cpp",
     "DefaultAclStorage.h",
+    "DefaultEnhancedSetupFlowProvider.cpp",
+    "DefaultEnhancedSetupFlowProvider.h",
+    "DefaultTermsAndConditionsProvider.cpp",
+    "DefaultTermsAndConditionsProvider.h",
```

Comment:
- ESF and T&C must be a source_set that is added as a dep to server ONLY if ESF is enabled in the product.
- Enabling should be based on args.gni and off by default
- See `config("server_config")` above and the `chip_app_use_echo` for an example of conditional values on build.

This is because we should not have to pay the cost of default implementations unless their are referenced and this is an optional feature. It's true that in many cases it's GC'd at link time, but that assumes there are no dependencies that just crop-up from someone else in teh future. Having a separatate source_set is the dependency management best practice we now use.